src/components/SearchPage/SearchPage.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Suspense, useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { useSwipeable } from "react-swipeable";
+import useVideoSearch from "../../hooks/useVideoSearch";
+import { Video } from "../../types/types";
+import Pagination from "../Pagination";
+import SearchBox from "../SearchBox";
+import "./SearchPage.css";
+
+const VideoList = React.lazy(() => import("../VideoList"));
+const VideoDetail = React.lazy(() => import("../VideoDetail"));
+
+function SearchPage() {
+  const [term, setTerm] = useState<string>("");
+  const [videos, setVideos] = useState<Video[]>([]);
+  const [selectedVideo, setSelectedVideo] = useState<Video | null>(null);
+  const [currentPage, setCurrentPage] = useState<number>(0);
+  const [pageCount, setPageCount] = useState<number>(1);
+  const [offset, setOffset] = useState<number>(4);
+  const [videosPerPage] = useState<number>(4);
```

Comment:
you don't need useState here if you have fixed number of cards, move VIDEOS_PER_PAGE constant out of component

Reply:
and probably set if as offset default value?